run.go

```diff
@@ -602,15 +606,24 @@ func setupApparmor(spec *specs.Spec, apparmorProfile string) error {
 	return nil
 }
 
-func setupTerminal(g *generate.Generator, terminalPolicy TerminalPolicy) {
+func setupTerminal(g *generate.Generator, terminalPolicy TerminalPolicy, terminalSize *specs.Box) {
 	switch terminalPolicy {
 	case DefaultTerminal:
-		g.SetProcessTerminal(terminal.IsTerminal(int(os.Stdout.Fd())))
+		onTerminal := terminal.IsTerminal(unix.Stdin) && terminal.IsTerminal(unix.Stdout) && terminal.IsTerminal(unix.Stderr)
```

Comment:
So if anyone is not a terminal you default everything to not being a terminal>

Reply:
Yeah.  Trying to get "not a terminal" when we're being invoked directly (so we can't use the `--tty` flag) is a guessing game otherwise.  This ends up changing the default for `build-using-dockerfile`, since there we always connect `stdin` to `/dev/null` for `RUN` instructions.

Reply:
ok